src/app/src/utils.js

```diff
@@ -106,3 +108,38 @@ export function getBoundaryShapeFilename(boundary) {
         boundary.utility.name.replaceAll(/\s+/g, '_'),
     ].join('_')}.geojson`;
 }
+
+export function getBoundaryPermissions({ boundary, user }) {
+    const permissions = {
+        canWrite: false,
+        canReview: false,
+        canApprove: false,
+    };
+
+    const status = boundary?.status;
+    const role = user?.role;
+
+    if (!status || !role) {
+        return permissions;
+    }
+
+    if (role === ROLES.ADMINISTRATOR) {
+        permissions.canWrite = true;
+        permissions.canReview = true;
+        permissions.canApprove = true;
+    }
+
+    if (role === ROLES.CONTRIBUTOR && status === BOUNDARY_STATUS.DRAFT) {
+        permissions.canWrite = true;
+    }
+
+    if (
+        role === ROLES.VALIDATOR &&
+        [BOUNDARY_STATUS.SUBMITTED, BOUNDARY_STATUS.IN_REVIEW].includes(status)
```

Comment:
This may be tweaked in the future. Perhaps we want them to be able to start a review, then mark it as approved? Or even if they've asked for revisions? Can be easily changed down the line.